dashboard/app/models/programming_environment.rb

```diff
@@ -44,12 +46,18 @@ def self.seed_all(glob="config/programming_environments/*.json")
   def self.seed_record(file_path)
     properties = properties_from_file(File.read(file_path))
     environment = ProgrammingEnvironment.find_or_initialize_by(name: properties[:name])
-    environment.update! properties
+    environment.update! properties.except(:categories)
+    environment.categories = properties[:categories].map do |category_config|
```

Comment:
Do you need to save environment after this?

Reply:
I _think_ not but I've been very confused by this lately. The only documentation I could find was from [guides.rubyonrails.org](https://guides.rubyonrails.org/association_basics.html#methods-added-by-has-many-collection-objects), which reads "The collection= method makes the collection contain only the supplied objects, by adding and deleting as appropriate. The changes are persisted to the database." So my understanding is this automatically persists this change to the database, which matches what I saw when testing

Reply:
This reminded me that I meant to add a controller test so I'll do that now